### DISCUSSION TITLE
partial hydration with useHydrateAtom

### DISCUSSION THREAD

**Discussion by aryankarim** [pmndrs/jotai#2603] [Q&A] [answered]
Hello guys,

My atom store is too big, and I want to be able to partially update an atom upon hydration like so:
```javascript
useHydrateAtoms([[bigObjectAtom,{  
key1:{...lotsOfStuff},
key2:{...lotsOfStuff},
key3:{
  key4: 'I wanna update only this key',
  ...lotsOfStuff
}
}]])

**Comment by dai-shi** [comment]
I think you need to separate the atom into smaller atoms.
If you use `useEffect` instead of `useHydrateAtom`, does it work for you? Maybe it can be a workaround.

**Reply by aryankarim** [reply]
The `useEffect` might work since the fetch result is not needed for the SSR'ed html. But if it was needed, sadly there is no partial hydration.   Should I propose this in an issue?

**Reply by dai-shi** [reply]
well, I'm not sure if I understand the real issue yet. Please continue discussion here.

Can you spot the bug/behavior with a _minimal_ reproduction without using the server? Just SPA. StackBlitz is preferable.

**Reply by aryankarim** [reply]
Here is an example where hydrating complex objects would becomes an untidy task:
That is just one scenario I have faced, I fixed it with having granular atoms.

[StackBlitz](https://stackblitz.com/edit/stackblitz-starters-i1r3mf)

If something like this would be possible, that would be nice:

```javascript
useHydrateAtoms([[complexAtom.nestedvalue, newValue]])

**Reply by dai-shi** [reply, marked as answer]
> an untidy task

Hm, so is it a DX issue? I'd create a function to initialize the atom value.
```js
const createComplexAtomValue = (number3, ssr_ed_setence) => (
      {
        numbers: {
          number1: 1,
          number2: 2,
          number3,
        },
        stenences: {
          sentence1: 'sentence1',
          sentence2: 'sentence2',
          sentence3: 'sentence3',
          sentence4: 'sentence4',
          sentence5: 'sentence5',
          ssr_ed_sentence,
        },
      },
);

const complexAtom = atom(createComplexAtomValue('default number', 'default sentence');
```

> I fixed it with having granular atoms.

Sounds like it's more Jotai way.

> If something like this would be possible

Unfortunately, I don't think it's possible.

**Reply by aryankarim** [reply]
Yeah, it is DX related. Thanks for the solution, that's is also another clean way to do it.
